pkg/uhttp/authcredentials_test.go

```diff
@@ -98,3 +98,35 @@ func TestHelpers_OAuth2_JWT_GetClient(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, client)
 }
+
+func TestHelpers_OAuth2_RefreshToken_GetClient(t *testing.T) {
+	rt := &OAuth2RefreshToken{
+		cfg: &oauth2.Config{
+			ClientID:     "test-client-id",
+			ClientSecret: "test-client-secret",
+			Endpoint: oauth2.Endpoint{
+				TokenURL: "https://test-token-url",
+			},
+		},
+		accessToken:  "test-access-token",
+		refreshToken: "test-refresh-token",
+	}
```

Comment:
I wonder if there's a service we could use to _actually_ test this but I guess it's pretty much a builtin for Go so maybe it's fine.

Reply:
Yeah, I don't think so, at least not easily accessible over long period of time, it might be worth to add mock server to test all of these flows.

Reply:
That would be great

Reply:
Just told @vlnevyhosteny about this and he would love to look into it. I'm going to merge this PR, so I can use it in ArubaCentral. 🙌🏻